doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+
+* patches required due to updates in the Zephyr main tree
+* urgent patches, for example those that fix security vulnerability issues,
+  and, therefore, should not wait to be fixed, first, in the external project.
+
+Non trivial changes to module code base, including changes in the module
+design or functionality should be discouraged, if the module has an upstream
+project repository. In that case, such changes shall be submitted to the
+upstream project, directly.
+
+:ref:`Submitting changes to modules <submitting_new_modules>` describes in
+detail the process of contributing changes to module repositories.
+
+Contribution guidelines
+-----------------------
+
+Contributing to Zephyr modules shall follow the generic project
+:ref:`Contribution guidelines <contribute_guidelines>`.
+
+**Pull Requests:** may be merged with minimum of 2 approvals, including
+an approval by the PR assignee. In addition to this, pull requests in module
+repositories may only be merged, if the introduced changes are verified
+with Zephyr CI tools, as described in more detail in other sections on
+this page.
+
+**Issue Reporting:** intentionally disabled in module repositories, in
+favor of a centralized policy for issue reporting. Tickets concerning, for
+example, bugs or enhancements in modules shall be opened in the main
+Zephyr repository. Issues should be appropriately labeled using Github
+labels corresponding to each module, when applicable.
+
+  .. note::
+
+     Bug reports in modules, mirroring upstream project bug reports, are
+     allowed but shall affect the
+     :ref:`Release Quality Criteria<release_quality_criteria>`.
+
+
+.. _modules_licensing:
+
+Licensing requirements and policies
+***********************************
+
+All source files in a module's code base require to include a license header,
+unless there exist a **main license file** that covers source files that do
+not possess a license header.
+
+Main license files shall be added in the module code base by Zephyr
+developers, only if they exist as part of the external project,
+and they contain a permissive OSI-compliant license. Main license files
+should preferably contain a full license text instead of including an
+SPDX license identifier. If multiple main license files are present it
+shall be made clear which license applies to each source file in a module
+code base.
+
+Individual license headers in module source files supersede the main license.
+
+Any new content to be added in a module repositotry will require to have
+license coverage.
+
+  .. note::
+
+     Zephyr recommends conveying module licensing via individual license
+     headers and main license files. This not a hard requirement; should
+     an external project have its own practice of conveying how licensing
+     applies in the module code base, this practice may be accepted by and
+     be referred to in the Zephyr module, as long as licensing requirement,
+     for example OSI compliance, are satisfied.
```

Comment:
This directly contradicts the above text, which says:

> All source files in a module's code base require to include a license header [...] unless the module repository has a **main license file**"

If the above text is true, then this **is** a hard requirement and this note is incorrect and should be fixed.

If it's false, then the above text must be fixed to make it clear that it's not required.

Reply:
Yeah, so the point here @mbolivar-nordic is that we should be flexible with the upstream project, e.g. accepting how they use root license files on their code bases. 
I will rephrase this text to make this more clear.

Reply:
Added 
```
(for example, by having a single or multiple main license files)
```

in the sentence about how the project can convey the licensing, to clearly indicate that it is about having one or more root license headers (and Zephyr should accept that).